### PR TITLE
DateTimeRange initial recipe

### PR DESCRIPTION
**Pull request by kthyng**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/DateTimeRange/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  imports:
+    - datetimerange
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/thombashi/DateTimeRange
+  summary: DateTimeRange is a Python library to handle a time range. e.g. check whether a time is within the time range, get the intersection of time ranges, truncating a time range, iterate through a time range, and so forth.
```
Comment (hmaarrfk) [inline recipes/DateTimeRange/meta.yaml:37]:
```suggestion
  summary: DateTimeRange is a Python library to handle a time range. 
  description: |
      Check whether a time is within the time range, get the intersection of time ranges, 
      truncating a time range, iterate through a time range, and so forth.
```

Summary is really supposed to be a 1 line thing

File: recipes/DateTimeRange/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  dev_url: https://github.com/thombashi/DateTimeRange
+  license: MIT
+  license_file:
+    - LICENSE
```
Comment (hmaarrfk) [inline recipes/DateTimeRange/meta.yaml:41]:
```suggestion
  license_file: LICENSE
```

File: recipes/DateTimeRange/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.6
+    - releasecmd
```
Comment (hmaarrfk) [inline recipes/DateTimeRange/meta.yaml:20]:
```suggestion
```
Do you really need releasecmd? it seems like it has a graceful failure if it isn't installed.

**Comment by hmaarrfk** [comment]
Please have the other maintainers verbally agree before we merge

**Comment by kthyng** [comment]
@hmaarrfk Thank you for your time looking at this! I'll admit I'm just a user of this fine package and not the author. I will ping the author to see about getting attention here: @thombashi are you ok with your package going into conda-forge? I also brought it up here: https://github.com/thombashi/DateTimeRange/issues/42

File: recipes/DateTimeRange/meta.yaml (hunk 3 above)
Comment (kthyng) [reply recipes/DateTimeRange/meta.yaml:20]: I'll wait to see what @thombashi thinks.

**Comment by hmaarrfk** [comment]
To be clear: 
I'm not so interested if thombashi "wants" it to go onto conda-forge, more if he "want to help maintain it".

I'm OK merging it with or without his "approval of getting it on conda-forge". But if he wants to help that is better.

This is because maintaining a package is additional work, and I don't want to impose it on anybody.

**Comment by kthyng** [comment]
@hmaarrfk Ah, I see. I put his handle in there without permission. Perhaps I should remove it if we don't hear back in a few days?

**Comment by hmaarrfk** [comment]
Their package is under the MIT license, you can do what you want with it so long as you abide by the license.

**Comment by kthyng** [comment]
@hmaarrfk Ok I'd like to plow ahead then if possible. Thanks again for your help.

**Pull request merged**